implementation/choreography-to-orchestration/lib/choreography-stack.ts

```diff
@@ -0,0 +1,512 @@
+            {
+              responseTemplates: {},
+              statusCode: "200",
+            },
+          ],
+          passthroughBehavior: "WHEN_NO_MATCH",
+          timeoutInMillis: 29000,
+          type: "AWS_PROXY",
+          uri: `arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/${ProcessPaymentFunction.functionArn}/invocations` 
+        },
+      }
+    );
+
+    const apiMethodRole2 = new iam.Role(this, "Api Method Role 2", {
+      assumedBy: new iam.ServicePrincipal("apigateway.amazonaws.com"),
+    });
+
+    apiMethodRole2.addToPolicy(
+      new iam.PolicyStatement({
+        actions: [
+          "dynamodb:BatchGetItem",
+          "dynamodb:Describe*",
+          "dynamodb:List*",
+          "dynamodb:GetItem",
+          "dynamodb:Query",
+          "dynamodb:Scan",
+        ],
+        resources: [DynamoDBTable.attrArn],
+      })
+    );
+
+    const ApiGatewayMethod2 = new apigateway.CfnMethod(
+      this,
+      "ApiGatewayMethod2",
+      {
+        restApiId: ApiGatewayRestApi.ref,
+        resourceId: ApiGatewayResource3.ref,
+        httpMethod: "GET",
+        authorizationType: "NONE",
+        apiKeyRequired: false,
+        requestParameters: {
+          "method.request.path.id": true,
+        },
+        methodResponses: [
+          {
+            responseModels: {
+              "application/json": "Empty",
+            },
+            statusCode: "200",
+          },
+        ],
+        integration: {
+          cacheNamespace: ApiGatewayResource3.ref,
+          credentials: apiMethodRole2.roleArn,
+          integrationHttpMethod: "POST",
+          integrationResponses: [
+            {
+              responseTemplates: {
+                "application/json": `{
+                    #set( $items= $input.path("$.Items") )
+                    #foreach( $item in $items )
+                        "Product Id" : $item.product_id.S,
+                        "Payment has been processed" : $item.Payment_processed.BOOL,
+                        "Order has been shipped" : $item.Ship_order.BOOL,
+                        "Reward was updated" : $item.Update_reward.BOOL
+                        #if( $foreach.hasNext ),#end
+                        $esc.newline
+                    #end
+                    }
+                    `,
+              },
+              statusCode: "200",
+            },
+          ],
+          passthroughBehavior: "WHEN_NO_TEMPLATES",
+          requestTemplates: {
+            "application/json": `{
+                "TableName": "store-order-data",
+                "KeyConditionExpression": "product_id=:v1",
+                "ExpressionAttributeValues": {
+                    ":v1": {
+                        "S": "$util.urlDecode($input.params('id'))"
+                    }
+                }
+            }`,
+          },
+          timeoutInMillis: 29000,
+          type: "AWS",
+          uri: "arn:aws:apigateway:us-east-1:dynamodb:action/Query",
```

Comment:
again, region is hardcoded and so this does not work when I am using us-west-2

Reply:
Added Dynamic parameter for region